src/nodes.coffee

```diff
@@ -2110,7 +2112,7 @@ exports.Param = class Param extends Base
     @reference = node
 
   isComplex: ->
-    @name.isComplex()
+    @name.isComplex() or @value instanceof Call
```

Comment:
Ugh, pity I didn't think of this sooner: this should probably just be `@name.isComplex() or @value.isComplex()`, the following is still broken:

```coffee
i = 0
({ a = ++i }, b = ++i) ->
```

It generates:

```js
// Generated by CoffeeScript 2.0.0-alpha
var i;

i = 0;

(function(arg, b = ++i) { // 'b' will be set first (b === 1)
  var a, ref;
  a = (ref = arg.a) != null ? ref : ++i; // 'a' will be set second (a === 2)
});
```